src/feature_extraction.py

```diff
@@ -335,11 +335,30 @@ def transform(self, X, **kwargs):
 
 
 class BureauFeatures(BasicHandCraftedFeatures):
+    def __init__(self, last_k_agg_periods, last_k_agg_period_fractions, num_workers=1, **kwargs):
+        super().__init__(num_workers=num_workers)
+        self.last_k_agg_periods = last_k_agg_periods
+        self.last_k_agg_period_fractions = last_k_agg_period_fractions
+
+        self.num_workers = num_workers
+        self.features = None
+
     def fit(self, bureau, **kwargs):
+        bureau.sort_values(['SK_ID_CURR', 'DAYS_CREDIT'], ascending=False, inplace=True)
         bureau['bureau_credit_active_binary'] = (bureau['CREDIT_ACTIVE'] != 'Closed').astype(int)
         bureau['bureau_credit_enddate_binary'] = (bureau['DAYS_CREDIT_ENDDATE'] > 0).astype(int)
-        features = pd.DataFrame({'SK_ID_CURR': bureau['SK_ID_CURR'].unique()})
+        bureau['bureau_credit_type_consumer'] = (bureau['CREDIT_TYPE'] == 'Consumer credit').astype(int)
```

Comment:
@kstrzala in reality a lot of those variablees are categoricals but in our outputs all of those fall under the `numerical_features` key. It is quite straighttforward to add 'categorical_features` key and output those variables there. Later one would also need to pass the bureau_features via categorical_features in feature_joiner.
What do you think?

Reply:
These properties are categorical if you view them as 'bureau' features, but right now they are added to the main feature table rather as numerical features-how many times one does apply for a certain credit, how does it compare to the total amount of credits etc. One can obviously try to extract some categorical info from that, but this is not the case here.